### PR TITLE
Add support for Python 3.12, ping proj back to 9.3.x, require geopandas>=1.0

### PR DESCRIPTION
**Pull request by jameslamb**
## Description

Contributes to https://github.com/rapidsai/build-planning/issues/40

This PR adds support for Python 3.12.

## Notes for Reviewers

This is part of ongoing work to add Python 3.12 support across RAPIDS.
It temporarily introduces a build/test matrix including Python 3.12, from https://github.com/rapidsai/shared-workflows/pull/213.

A follow-up PR will revert back to pointing at the `branch-24.10` branch of `shared-workflows` once all
RAPIDS repos have added Python 3.12 support.

Other changes required to add that support:

* pinning `proj` back to 9.3.x, reverting #1449 (https://github.com/rapidsai/cuspatial/pull/1453#issuecomment-2335527542)
* requiring `geopandas >= 1.0` (https://github.com/rapidsai/cuspatial/pull/1453#discussion_r1750444386)

### This will fail until all dependencies have been updates to Python 3.12

CI here is expected to fail until all of this project's upstream dependencies support Python 3.12.

This can be merged whenever all CI jobs are passing.

### CODE REVIEW

**Comment by jameslamb** [comment]
Some changes to the required version of `gdal`:

```text
LibMambaUnsatisfiableError: Encountered problems while solving:
  - nothing provides gdal 1.11.4 needed by fiona-1.6.3-np110py27_0
```

Working on it.

**Comment by jameslamb** [comment]
😬 😬 😬 I think being behind on `fmt` and `spdlog` (https://github.com/rapidsai/build-planning/issues/56#issuecomment-2288846888) is finally catching up to us here.

I haven't 100% figured this out yet, but posting my thoughts so far.

## Short Summary

Here's what I think is happening:

* several geospatial dependencies require `gdal` / `libgdal` at runtime
* `libgdal` depends on `tiledb`
* `tiledb` first introduced a runtime dependency on `fmt` and `spdlog` in `tiledb==2.20.1` (https://github.com/conda-forge/tiledb-feedstock/pull/254)
* `librmm` is a run dependency of `cuspatial` (directly and transitively, via `cudf`)
* `librmm` has these runtime constraints: `fmt >=10.1.1,<11, spdlog >=1.12.0,<1.13`
* the newest version of `tiledb` on conda-forge compatible with those is `tiledb==2.22.0`
* with `python=3.12`, something is preventing conda from falling back to an old enough `gdal` / `libgdal` that it can also install `tiledb<=2.22.0`

If I'm right, options I think we have:

* upgrade `fmt` and `spdlog` across RAPIDS (https://github.com/rapidsai/build-planning/issues/56)
* remove `run` / `run_exports` dependencies on `librmm` in RAPIDS conda packages (https://github.com/rapidsai/build-planning/issues/92)
  - *that'd at least remove a source of `fmt` / `spdlog` run dependencies and improve the success rate for runtime / test-time solves like this... but not sure what other implication it'd have*

## Other Notes

### How to reproduce (and then fiddle with) the exact errors from CI

<details><summary>commands to run (click me)</summary>

On an x86_64 machine with CUDA

```shell
docker run \
  --rm \
  --gpus 1 \
  --env RAPIDS_BUILD_TYPE=pull-request \
  --env RAPIDS_REPOSITORY=rapidsai/cuspatial \
  --env RAPIDS_REF_NAME=pull-request/1453 \
  -v $(pwd):/opt/work \
  -w /opt/work \
  -it rapidsai/ci-conda:cuda12.5.1-ubuntu22.04-py3.12 \
  bash

CPP_CHANNEL=$(rapids-download-conda-from-s3 cpp)
PYTHON_CHANNEL=$(rapids-download-conda-from-s3 python)

# guarantee that exactly the CI artifacts are used
# and that 'mamba install' doesn't silently fall back to
# other packages from rapidsai-nightly / rapidsai
BUILD_SUFFIX="240906_g4361f9ed_42"

mamba create \
  --name delete-me \
  -c conda-forge \
  -c "${CPP_CHANNEL}" \
  -c "${PYTHON_CHANNEL}" \
  --dry-run \
  "cuproj==24.10.00a42=cuda12_py312_${BUILD_SUFFIX}" \
  "cuspatial==24.10.00a42=cuda12_py312_${BUILD_SUFFIX}" \
  "libcuspatial==24.10.00a42=cuda12_${BUILD_SUFFIX}" \
  "geopandas>=0.11.0" \
  "python=3.12" 
```

</details>

<details><summary>full solver output (click me)</summary>

```text
The following packages are incompatible
├─ cuproj ==24.10.00a42 cuda12_py312_240906_g4361f9ed_42 is installable and it requires
│  ├─ librmm >=24.10.0a.32,<24.11.0a0 , which requires
│  │  ├─ fmt >=10.1.1,<11 , which can be installed;
│  │  └─ spdlog >=1.12.0,<1.13  with the potential options
│  │     ├─ spdlog 1.12.0 would require
│  │     │  └─ fmt >=9.1.0,<10.0a0 , which conflicts with any installable versions previously reported;
│  │     └─ spdlog 1.12.0, which can be installed;
│  └─ proj >=9.4.1,<9.5.0a0 , which can be installed;
├─ cuspatial ==24.10.00a42 cuda12_py312_240906_g4361f9ed_42 is not installable because it requires
│  ├─ geopandas >=0.11.0  with the potential options
│  │  ├─ geopandas [0.11.0|0.11.1|0.12.0] would require
│  │  │  └─ geopandas-base [0.11.0 pyha770c72_0|0.11.1 pyha770c72_0|0.12.0 pyha770c72_0], which requires
│  │  │     └─ shapely >=1.7,<2  with the potential options
│  │  │        ├─ shapely 1.7.0 would require
│  │  │        │  └─ python >=2.7,<2.8.0a0 , which can be installed;
│  │  │        ├─ shapely 1.7.0 would require
│  │  │        │  └─ python_abi 2.7.* *_cp27mu, which can be installed;
│  │  │        ├─ shapely [1.7.0|1.7.1] would require
│  │  │        │  └─ python_abi 3.6.* *_cp36m, which can be installed;
│  │  │        ├─ shapely 1.7.0 would require
│  │  │        │  └─ python >=3.6,<3.7.0a0 , which can be installed;
│  │  │        ├─ shapely 1.7.0 would require
│  │  │        │  └─ python >=3.7,<3.8.0a0 , which can be installed;
│  │  │        ├─ shapely [1.7.0|1.7.1|...|1.8.5] would require
│  │  │        │  └─ python_abi 3.7.* *_cp37m, which can be installed;
│  │  │        ├─ shapely 1.7.0 would require
│  │  │        │  └─ python >=3.8,<3.9.0a0 , which can be installed;
│  │  │        ├─ shapely [1.7.0|1.7.1|...|1.8.5] would require
│  │  │        │  └─ python_abi 3.8.* *_cp38, which can be installed;
│  │  │        ├─ shapely 1.7.1 would require
│  │  │        │  └─ python_abi 3.6 *_pypy36_pp73, which can be installed;
│  │  │        ├─ shapely [1.7.1|1.8.0] would require
│  │  │        │  └─ python_abi 3.7 *_pypy37_pp73, which can be installed;
│  │  │        ├─ shapely [1.7.1|1.8.0|...|1.8.5] would require
│  │  │        │  └─ python_abi 3.9.* *_cp39, which can be installed;
│  │  │        ├─ shapely [1.8.0|1.8.1|...|1.8.5] would require
│  │  │        │  └─ python_abi 3.10.* *_cp310, which can be installed;
│  │  │        ├─ shapely [1.8.0|1.8.1|...|1.8.5] would require
│  │  │        │  └─ python_abi 3.8 *_pypy38_pp73, which can be installed;
│  │  │        ├─ shapely [1.8.0|1.8.1|...|1.8.5] would require
│  │  │        │  └─ python_abi 3.9 *_pypy39_pp73, which can be installed;
│  │  │        └─ shapely 1.8.5 would require
│  │  │           └─ python_abi 3.11.* *_cp311, which can be installed;
│  │  ├─ geopandas [0.12.1|0.12.2|0.13.0|0.13.1|0.13.2] would require
│  │  │  └─ fiona with the potential options
│  │  │     ├─ fiona [1.6.3|1.6.4] would require
│  │  │     │  └─ gdal 1.11.4 , which does not exist (perhaps a missing channel);
│  │  │     ├─ fiona [1.7.0|1.8.11|...|1.8.9.post2] would require
│  │  │     │  └─ python [2.7* |>=2.7,<2.8.0a0 ], which can be installed;
│  │  │     ├─ fiona [1.7.0|1.7.1|...|1.7.9] would require
│  │  │     │  └─ gdal 2.1.* , which does not exist (perhaps a missing channel);
│  │  │     ├─ fiona 1.7.0 would require
│  │  │     │  └─ python 3.4* , which can be installed;
│  │  │     ├─ fiona 1.7.0 would require
│  │  │     │  └─ python 3.5* , which can be installed;
│  │  │     ├─ fiona [1.7.11|1.7.12|1.7.13] would require
│  │  │     │  └─ gdal 2.2.* , which does not exist (perhaps a missing channel);
│  │  │     ├─ fiona 1.7.13 would require
│  │  │     │  └─ gdal >=2.2.2,<2.3.0a0 , which does not exist (perhaps a missing channel);
│  │  │     ├─ fiona 1.7.13 would require
│  │  │     │  └─ gdal >=2.2.4,<2.3.0a0 , which does not exist (perhaps a missing channel);
│  │  │     ├─ fiona [1.7.8|1.7.9] would require
│  │  │     │  └─ libgdal 2.1.4 , which does not exist (perhaps a missing channel);
│  │  │     ├─ fiona [1.8.0|1.8.1|1.8.2|1.8.3|1.8.4] would require
│  │  │     │  └─ gdal >=2.3.2,<2.4.0a0 , which does not exist (perhaps a missing channel);
│  │  │     ├─ fiona [1.8.11|1.8.12|...|1.8.9.post2] would require
│  │  │     │  └─ python >=3.6,<3.7.0a0 , which can be installed;
│  │  │     ├─ fiona [1.8.11|1.8.12|...|1.8.9.post2] would require
│  │  │     │  └─ python >=3.7,<3.8.0a0 , which can be installed;
│  │  │     ├─ fiona [1.8.11|1.8.12|1.8.13|1.8.9.post2] would require
│  │  │     │  └─ python >=3.8,<3.9.0a0 , which can be installed;
│  │  │     ├─ fiona [1.8.13|1.8.15|...|1.8.20] would require
│  │  │     │  └─ python_abi 3.6.* *_cp36m, which can be installed;
│  │  │     ├─ fiona [1.8.13|1.8.15|...|1.8.22] would require
│  │  │     │  └─ python_abi 3.7.* *_cp37m, which can be installed;
│  │  │     ├─ fiona [1.8.13|1.8.15|...|1.9.6] would require
│  │  │     │  └─ python_abi 3.8.* *_cp38, which can be installed;
│  │  │     ├─ fiona [1.8.17|1.8.18|...|1.9.6] would require
│  │  │     │  └─ python_abi 3.9.* *_cp39, which can be installed;
│  │  │     ├─ fiona [1.8.20|1.8.21|...|1.9.6] would require
│  │  │     │  └─ python_abi 3.10.* *_cp310, which can be installed;
│  │  │     ├─ fiona [1.8.22|1.9.0|...|1.9.6] would require
│  │  │     │  └─ python_abi 3.11.* *_cp311, which can be installed;
│  │  │     ├─ fiona 1.10.0 would require
│  │  │     │  ├─ libgdal >=3.9.2,<3.10.0a0 , which requires
│  │  │     │  │  ├─ libgdal-tiledb 3.9.2.* , which requires
│  │  │     │  │  │  └─ tiledb >=2.25.0,<2.26.0a0  but there are no viable options
│  │  │     │  │  │     ├─ tiledb [2.23.0|2.23.1|...|2.25.0] would require
│  │  │     │  │  │     │  ├─ fmt >=10.2.1,<11.0a0 , which can be installed;
│  │  │     │  │  │     │  └─ spdlog >=1.13.0,<1.14.0a0 , which conflicts with any installable versions previously reported;
│  │  │     │  │  │     ├─ tiledb [2.24.2|2.25.0] would require
│  │  │     │  │  │     │  └─ fmt >=11.0.2,<12.0a0 , which conflicts with any installable versions previously reported;
│  │  │     │  │  │     └─ tiledb [2.24.2|2.25.0] would require
│  │  │     │  │  │        └─ fmt >=11.0.1,<12.0a0 , which conflicts with any installable versions previously reported;
│  │  │     │  │  └─ tiledb >=2.25.0,<2.26.0a0 , which cannot be installed (as previously explained);
│  │  │     │  └─ python_abi 3.10.* *_cp310, which can be installed;
│  │  │     ├─ fiona 1.10.0 would require
│  │  │     │  ├─ libgdal >=3.9.2,<3.10.0a0 , which cannot be installed (as previously explained);
│  │  │     │  └─ python_abi 3.11.* *_cp311, which can be installed;
│  │  │     ├─ fiona 1.10.0 would require
│  │  │     │  └─ libgdal >=3.9.2,<3.10.0a0 , which cannot be installed (as previously explained);
│  │  │     ├─ fiona 1.10.0 would require
│  │  │     │  ├─ libgdal >=3.9.2,<3.10.0a0 , which cannot be installed (as previously explained);
│  │  │     │  └─ python_abi 3.9.* *_cp39, which can be installed;
│  │  │     ├─ fiona [1.9.4|1.9.5] would require
│  │  │     │  └─ libgdal >=3.7.2,<3.8.0a0  but there are no viable options
│  │  │     │     ├─ libgdal [3.7.2|3.7.3|3.8.0|3.8.1] would require
│  │  │     │     │  └─ proj >=9.3.0,<9.3.1.0a0 , which conflicts with any installable versions previously reported;
│  │  │     │     ├─ libgdal 3.7.2 would require
│  │  │     │     │  └─ proj >=9.2.1,<9.2.2.0a0 , which conflicts with any installable versions previously reported;
│  │  │     │     ├─ libgdal [3.7.3|3.8.1|3.8.2|3.8.3|3.8.4] would require
│  │  │     │     │  └─ proj >=9.3.1,<9.3.2.0a0 , which conflicts with any installable versions previously reported;
│  │  │     │     ├─ libgdal [3.7.3|3.8.4|3.8.5|3.9.0] would require
│  │  │     │     │  └─ proj >=9.4.0,<9.4.1.0a0 , which conflicts with any installable versions previously reported;
│  │  │     │     └─ libgdal [3.7.3|3.8.5|3.9.0] would require
│  │  │     │        ├─ proj >=9.4.0,<9.4.1.0a0 , which conflicts with any installable versions previously reported;
│  │  │     │        └─ tiledb >=2.23.0,<2.24.0a0 , which cannot be installed (as previously explained);
│  │  │     ├─ fiona 1.9.5 would require
│  │  │     │  └─ libgdal >=3.8.0,<3.9.0a0  but there are no viable options
│  │  │     │     ├─ libgdal [3.7.2|3.7.3|3.8.0|3.8.1], which cannot be installed (as previously explained);
│  │  │     │     ├─ libgdal [3.7.3|3.8.1|3.8.2|3.8.3|3.8.4], which cannot be installed (as previously explained);
│  │  │     │     ├─ libgdal [3.7.3|3.8.4|3.8.5|3.9.0], which cannot be installed (as previously explained);
│  │  │     │     ├─ libgdal [3.7.3|3.8.5|3.9.0], which cannot be installed (as previously explained);
│  │  │     │     ├─ libgdal [3.8.5|3.9.0|3.9.1] would require
│  │  │     │     │  ├─ libgdal-tiledb 3.9.1.*  but there are no viable options
│  │  │     │     │  │  ├─ libgdal-tiledb [3.9.1|3.9.2], which cannot be installed (as previously explained);
│  │  │     │     │  │  └─ libgdal-tiledb 3.9.1 would require
│  │  │     │     │  │     └─ tiledb >=2.24.2,<2.25.0a0  but there are no viable options
│  │  │     │     │  │        ├─ tiledb [2.23.0|2.23.1|...|2.25.0], which cannot be installed (as previously explained);
│  │  │     │     │  │        ├─ tiledb [2.24.2|2.25.0], which cannot be installed (as previously explained);
│  │  │     │     │  │        ├─ tiledb [2.24.2|2.25.0], which cannot be installed (as previously explained);
│  │  │     │     │  │        └─ tiledb 2.24.2 would require
│  │  │     │     │  │           ├─ fmt >=10.2.1,<11.0a0 , which can be installed;
│  │  │     │     │  │           └─ spdlog >=1.14.1,<1.15.0a0  but there are no viable options
│  │  │     │     │  │              ├─ spdlog 1.14.1 conflicts with any installable versions previously reported;
│  │  │     │     │  │              └─ spdlog 1.14.1 would require
│  │  │     │     │  │                 └─ fmt >=11.0.1,<12.0a0 , which conflicts with any installable versions previously reported;
│  │  │     │     │  └─ tiledb [>=2.24.0,<2.25.0a0 |>=2.24.1,<2.25.0a0 |>=2.24.2,<2.25.0a0 ], which cannot be installed (as previously explained);
│  │  │     │     ├─ libgdal [3.8.5|3.9.2], which cannot be installed (as previously explained);
│  │  │     │     └─ libgdal [3.8.5|3.9.0] would require
│  │  │     │        └─ tiledb >=2.23.0,<2.24.0a0 , which cannot be installed (as previously explained);
│  │  │     ├─ fiona [1.9.5|1.9.6] would require
│  │  │     │  └─ libgdal [>=3.8.2,<3.9.0a0 |>=3.8.4,<3.9.0a0 ], which cannot be installed (as previously explained);
│  │  │     ├─ fiona 1.9.6 would require
│  │  │     │  ├─ libgdal >=3.9.1,<3.10.0a0  but there are no viable options
│  │  │     │  │  ├─ libgdal [3.8.5|3.9.0|3.9.1], which cannot be installed (as previously explained);
│  │  │     │  │  ├─ libgdal [3.8.5|3.9.2], which cannot be installed (as previously explained);
│  │  │     │  │  └─ libgdal 3.9.1 would require
│  │  │     │  │     └─ libgdal-tiledb 3.9.1.* , which cannot be installed (as previously explained);
│  │  │     │  └─ python_abi 3.10.* *_cp310, which can be installed;
│  │  │     ├─ fiona 1.9.6 would require
│  │  │     │  ├─ libgdal >=3.9.1,<3.10.0a0 , which cannot be installed (as previously explained);
│  │  │     │  └─ python_abi 3.11.* *_cp311, which can be installed;
│  │  │     ├─ fiona 1.9.6 would require
│  │  │     │  └─ libgdal [>=3.8.5,<3.9.0a0 |>=3.9.0,<3.10.0a0 ], which cannot be installed (as previously explained);
│  │  │     ├─ fiona 1.9.6 would require
│  │  │     │  └─ libgdal >=3.9.1,<3.10.0a0 , which cannot be installed (as previously explained);
│  │  │     ├─ fiona 1.9.6 would require
│  │  │     │  ├─ libgdal >=3.9.1,<3.10.0a0 , which cannot be installed (as previously explained);
│  │  │     │  └─ python_abi 3.8.* *_cp38, which can be installed;
│  │  │     └─ fiona 1.9.6 would require
│  │  │        ├─ libgdal >=3.9.1,<3.10.0a0 , which cannot be installed (as previously explained);
│  │  │        └─ python_abi 3.9.* *_cp39, which can be installed;
│  │  ├─ geopandas 0.13.2 would require
│  │  │  └─ fiona >=1.8.19  with the potential options
│  │  │     ├─ fiona [1.8.13|1.8.15|...|1.8.20], which can be installed (as previously explained);
│  │  │     ├─ fiona [1.8.13|1.8.15|...|1.8.22], which can be installed (as previously explained);
│  │  │     ├─ fiona [1.8.13|1.8.15|...|1.9.6], which can be installed (as previously explained);
│  │  │     ├─ fiona [1.8.17|1.8.18|...|1.9.6], which can be installed (as previously explained);
│  │  │     ├─ fiona [1.8.20|1.8.21|...|1.9.6], which can be installed (as previously explained);
│  │  │     ├─ fiona [1.8.22|1.9.0|...|1.9.6], which can be installed (as previously explained);
│  │  │     ├─ fiona 1.10.0, which cannot be installed (as previously explained);
│  │  │     ├─ fiona 1.10.0, which cannot be installed (as previously explained);
│  │  │     ├─ fiona 1.10.0, which cannot be installed (as previously explained);
│  │  │     ├─ fiona 1.10.0, which cannot be installed (as previously explained);
│  │  │     ├─ fiona [1.9.4|1.9.5], which cannot be installed (as previously explained);
│  │  │     ├─ fiona 1.9.5, which cannot be installed (as previously explained);
│  │  │     ├─ fiona [1.9.5|1.9.6], which cannot be installed (as previously explained);
│  │  │     ├─ fiona 1.9.6, which cannot be installed (as previously explained);
│  │  │     ├─ fiona 1.9.6, which cannot be installed (as previously explained);
│  │  │     ├─ fiona 1.9.6, which cannot be installed (as previously explained);
│  │  │     ├─ fiona 1.9.6, which cannot be installed (as previously explained);
│  │  │     ├─ fiona 1.9.6, which cannot be installed (as previously explained);
│  │  │     └─ fiona 1.9.6, which cannot be installed (as previously explained);
│  │  ├─ geopandas [0.14.0|0.14.1|0.14.2|0.14.3|0.14.4] would require
│  │  │  └─ fiona >=1.8.21  with the potential options
│  │  │     ├─ fiona [1.8.13|1.8.15|...|1.8.22], which can be installed (as previously explained);
│  │  │     ├─ fiona [1.8.13|1.8.15|...|1.9.6], which can be installed (as previously explained);
│  │  │     ├─ fiona [1.8.17|1.8.18|...|1.9.6], which can be installed (as previously explained);
│  │  │     ├─ fiona [1.8.20|1.8.21|...|1.9.6], which can be installed (as previously explained);
│  │  │     ├─ fiona [1.8.22|1.9.0|...|1.9.6], which can be installed (as previously explained);
│  │  │     ├─ fiona 1.10.0, which cannot be installed (as previously explained);
│  │  │     ├─ fiona 1.10.0, which cannot be installed (as previously explained);
│  │  │     ├─ fiona 1.10.0, which cannot be installed (as previously explained);
│  │  │     ├─ fiona 1.10.0, which cannot be installed (as previously explained);
│  │  │     ├─ fiona [1.9.4|1.9.5], which cannot be installed (as previously explained);
│  │  │     ├─ fiona 1.9.5, which cannot be installed (as previously explained);
│  │  │     ├─ fiona [1.9.5|1.9.6], which cannot be installed (as previously explained);
│  │  │     ├─ fiona 1.9.6, which cannot be installed (as previously explained);
│  │  │     ├─ fiona 1.9.6, which cannot be installed (as previously explained);
│  │  │     ├─ fiona 1.9.6, which cannot be installed (as previously explained);
│  │  │     ├─ fiona 1.9.6, which cannot be installed (as previously explained);
│  │  │     ├─ fiona 1.9.6, which cannot be installed (as previously explained);
│  │  │     └─ fiona 1.9.6, which cannot be installed (as previously explained);
│  │  └─ geopandas [1.0.0|1.0.1] would require
│  │     └─ pyogrio >=0.7.2  with the potential options
│  │        ├─ pyogrio [0.7.2|0.8.0|0.9.0] would require
│  │        │  └─ python_abi 3.10.* *_cp310, which can be installed;
│  │        ├─ pyogrio [0.7.2|0.8.0|0.9.0] would require
│  │        │  └─ python_abi 3.11.* *_cp311, which can be installed;
│  │        ├─ pyogrio 0.7.2 would require
│  │        │  └─ libgdal >=3.8.0,<3.9.0a0 , which cannot be installed (as previously explained);
│  │        ├─ pyogrio 0.7.2 would require
│  │        │  └─ libgdal >=3.7.2,<3.8.0a0 , which cannot be installed (as previously explained);
│  │        ├─ pyogrio [0.7.2|0.8.0|0.9.0] would require
│  │        │  └─ python_abi 3.8.* *_cp38, which can be installed;
│  │        ├─ pyogrio [0.7.2|0.8.0|0.9.0] would require
│  │        │  └─ python_abi 3.9.* *_cp39, which can be installed;
│  │        ├─ pyogrio [0.8.0|0.9.0] would require
│  │        │  └─ libgdal [>=3.8.5,<3.9.0a0 |>=3.9.0,<3.10.0a0 ], which cannot be installed (as previously explained);
│  │        ├─ pyogrio 0.9.0 would require
│  │        │  └─ libgdal >=3.9.2,<3.10.0a0 , which cannot be installed (as previously explained);
│  │        └─ pyogrio 0.9.0 would require
│  │           └─ python >=3.13.0rc1,<3.14.0a0 , which requires
│  │              └─ _python_rc, which does not exist (perhaps a missing channel);
│  └─ python >=3.12,<3.13.0a0 , which requires
│     └─ python_abi 3.12.* *_cp312, which conflicts with any installable versions previously reported;
└─ libcuspatial ==24.10.00a42 cuda12_240906_240906_g4361f9ed_42 does not exist (perhaps a typo or a missing channel).
```

</details>

### Other notes

* `geopandas` dropped its `fiona` dependency (on conda-forge) in v1.0.0: https://github.com/conda-forge/geopandas-feedstock/pull/116
* `tiledb` has run dependencies on `fmt` and `spdlog`, and has participated in the recent migrations for those on conda-forge.
  - https://github.com/conda-forge/tiledb-feedstock/pull/324
  - https://github.com/conda-forge/tiledb-feedstock/pull/326
  - https://github.com/conda-forge/tiledb-feedstock/pull/327

**Comment by bdice** [comment]
Yup, I agree with your assessment. Here's a screenshot that also points to spdlog/fmt issues: 
image

This is from the command
```
mamba create -n test --dry-run -c rapidsai-nightly -c conda-forge cuspatial==24.10.00a* proj==9.4.1 cuda-version==12.5 python==3.11
```

Our other choice is to pin `cuproj` to use `proj>=9.3.0,<9.3.1.0a0`. That's what I'm going to propose for this PR since it confines the problem to the cuspatial/cuproj repository.

**Comment by bdice** [comment]
The `proj` pinning helped. I fixed an issue with the new C++ wheel workflows which weren't using `shared-workflows` with Python 3.12. This should pass CI now.

**Comment by jameslamb** [comment]
Ah thank you so much! I hadn't been able to isolate `proj` as something we could pin back to be able to use older gdal, tiledb, etc. 

With that change, you can now see conda able to fall back to older versions of those things:

```text
fmt                       10.2.1               h00ab1b0_0    conda-forge
...
gdal                      3.8.1           py312ha82c271_3    conda-forge
...
spdlog                    1.12.0               hd2e6256_2    conda-forge
...
tiledb                    2.18.2               h99f50a1_1    conda-forge
```

([build link](https://github.com/rapidsai/cuspatial/actions/runs/10773701421/job/29875056709?pr=1453))

Now we just have one other issue... the version of `fiona` this project builds from source in its CI fails to compile with Python 3.12.

```text
      fiona/schema.c: In function ‘__Pyx_PyIndex_AsSsize_t’:
      fiona/schema.c:5193:45: error: ‘PyLongObject’ {aka ‘struct _longobject’} has no member named ‘ob_digit’
       5193 |     const digit* digits = ((PyLongObject*)b)->ob_digit;
```

([build link](https://github.com/rapidsai/cuspatial/actions/runs/10773701421/job/29874993872?pr=1453))

`fiona==1.10.0` has Python 3.12 wheels available (https://pypi.org/project/fiona/#files), so we could try moving the pin to that. We'll still have to build from source in CI, I think, because I don't see aarch64 wheels there.

HOWEVER... first, I'm going to try just moving `cuspatial` to `geopandas>=1.0`. It was temporarily pinned in tests to `geopandas<1` in https://github.com/rapidsai/cuspatial/pull/1400#discussion_r1669398538, but I think that we may be able to remove that ceiling now that #1403 has been addressed.

This helps becuase `geopandas` dropped its `fiona` dependency in the 1.x series: https://github.com/conda-forge/geopandas-feedstock/pull/116

File: README.md [hunk 1]
```diff
@@ -145,42 +145,6 @@ pip install cuspatial-cu12 --extra-index-url=https://pypi.nvidia.com
 pip install cuspatial-cu11 --extra-index-url=https://pypi.nvidia.com
 ```
 
-#### Troubleshooting Fiona/GDAL versions
```
[the remark this answers is not in the document]
Comment (jameslamb) [reply README.md:148]: (cc @mroeschke )

**Review by bdice — APPROVED** [review]
All changes look good to me!

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (mroeschke) [reply README.md:148]: I don't have the strongest opinion on `geopandas` support, but given your breakdown, +1 for requiring `>= 1.0.0`

[the remark this answers is not in the document]
Comment (jameslamb) [reply README.md:148]: Thanks very much!

Ok, I'm going to merge this as-is then. @harrism `@`-ing you for awareness... if you have any concerns about this, let me know and I'd be happy to address them in a follow-up PR.

**Pull request merged**